### ISSUE TITLE
Problem when sending

### ISSUE THREAD

**Issue by Elektryon-Dev** [ombre-projects/ombre#47]
It is not a clear way to reproduce this. usually it comes when huge payment needs to be done.
2018-04-23 06:51:41 Error with transfer RPC request to wallet daemon {"code":-4,"message":"outs.size() < out_index + 1"}

**Comment by Bathmat** [comment]
I think I have a similar issue: `error::get_random_outs_error`

`"code":-4,"message":"failed to get random outs"`

Also when using the cli: `Error: failed to get random outputs to mix`

Not sure how to fix this...

**Comment by Bathmat** [comment]
Any thoughts about this issue? It's making it difficult to send transactions, even when using the cli.

I've tried just about every version of the wallet. `ombred` is running version 1.4 downloaded from the releases.

I've been trying to send all the OMB from one wallet to another and I keep having to decrease the amount that the walle-cli will send without getting the `Error: failed to get random outputs to mix`

`sweep_all` also gives the same error.

`balance detail` shows:

```
Balance: 374.988208963, unlocked balance: 355.047618894
Balance per address:
           Address               Balance      Unlocked balance  Outputs                 Label
       0 cashBWpwr         374.988208963         355.047618894      224       Primary account
```
So, is it an issue with having too many outputs?

**Comment by Bluecoreg** [comment]
This pull request should have fixed this issue.
https://github.com/ombre-projects/ombre/pull/34

I'll look into it. In the meantime do not send more than 100,000 Ombre per transaction.

**Comment by Bathmat** [comment]
I can't send more than 150 OMB per tx using the rpc, and that wallet above will only send 10 OMB per tx using the cli without getting the `Error: failed to get random outputs to mix`

**Comment by Bathmat** [comment]
Is this code correct?
```
    if (version == 2) {
      size_t tx_size_limit = TRANSACTION_SIZE_LIMIT;
    } else {
      size_t tx_size_limit = TRANSACTION_SIZE_LIMIT_V2;
    }
```
It's in that commit you linked @Bluecoreg

**Comment by Bathmat** [comment]
Ok, I think I've fixed this (at least for now)... I had recently switched the daemon over to run as service rather than just in a screen... When I did that, I added the flags `--rpc-bind-ip 127.0.0.1` and `--restricted-rpc`. Removing those seems to have fixed the errors. Not sure which was causing the problem, but my guess is `--restricted-rpc` was limiting the txs somehow.

What's odd is that it would work for a bit (about a day), then it would start giving me `random_outs` error.
